Tests/libhostmgrTests/GitMirrorTests.swift

```diff
@@ -0,0 +1,65 @@
+import Foundation
+import TSCBasic
+import XCTest
+import libhostmgr
+
+class GitMirrorTests: XCTestCase {
+
+    func testManifestAreSorted() throws {
+        let manifest = try getManifest(withFiles: [
+            "automattic/day-one-apple/2023-03-28.git.tar",
+            "automattic/pocket-casts-ios/2023-03-28.git.tar",
+            "automattic/simplenote-ios/2023-03-28.git.tar"
+        ])
+
+        XCTAssertEqual(
+            manifest,
+            """
+            automattic/day-one-apple/2023-03-28.git.tar
+            automattic/pocket-casts-ios/2023-03-28.git.tar
+            automattic/simplenote-ios/2023-03-28.git.tar
+            """
+        )
+    }
+
+    func testManifestOnlyContainsTarFiles() throws {
+        let manifest = try getManifest(withFiles: [
+            "automattic/day-one-apple/2023-03-28.git.tar",
+            "automattic/day-one-apple/na",
+            "automattic/pocket-casts-ios/2023-03-28.git.tar",
+            "automattic/pocket-casts-ios/nah",
+            "automattic/simplenote-ios/2023-03-28.git.tar"
+        ])
+        XCTAssertEqual(
+            manifest,
+            """
+            automattic/day-one-apple/2023-03-28.git.tar
+            automattic/pocket-casts-ios/2023-03-28.git.tar
+            automattic/simplenote-ios/2023-03-28.git.tar
+            """
+        )
+    }
+
+    func testManifestOnlyContainsTheMostRecentOne() throws {
+        let manifest = try getManifest(withFiles: [
+            "automattic/day-one-apple/2023-03-27.git.tar",
+            "automattic/day-one-apple/2023-03-28.git.tar"
+        ])
+        XCTAssertEqual(manifest, "automattic/day-one-apple/2023-03-28.git.tar")
+    }
+
+    private func getManifest(withFiles files: [String]) throws -> String {
+        let gitMirrorDir = AbsolutePath("/git-mirrors")
+        let fileSystem = InMemoryFileSystem()
```

Comment:
TIL. Really cool to have that concept of an `InMemoryFileSystem` for mocking, nice!